packages/babel-traverse/src/path/modification.js

```diff
@@ -112,7 +116,10 @@ export function insertAfter(nodes) {
     this.parentPath.isExpressionStatement() ||
     this.parentPath.isLabeledStatement()
   ) {
-    return this.parentPath.insertAfter(nodes);
+    // `replaceWithMultiple` requeues if there's a replacement for this.node,
+    // but not for an ancestor's. Set `shouldRequeue` to true so that any replacement
+    // for an ancestor node can be enqueued. Fix #5628 and #5023.
+    return this.parentPath._insertAfter(nodes, true);
```

Comment:
Good point. I may miss the mark, that's probably a separate bug. If we introduce that same logic in `insertBefore`, I think the example you posted will still fail because it wouldn't go into the branch `parentPath.insertBefore`. The current fix works based on the fact that `replaceWithMultiple` will requeue when replacement happens at current path, but not ancestor path, so only need to add requeuing when replacement happens at ancestor path. If we expect requeuing to happen when directly invoking `insertBefore` or `insertAfter`, maybe the solution is always invoke `this.requeue()` right after `this._replaceWith()`? My concern is this means a double requeuing when invoking `replaceWithMultiple()` and replacement happens at the current path. What do you think?